picard/__init__.py

```diff
@@ -87,3 +87,48 @@ def version_from_string(version_str):
 ]
 
 api_versions_tuple = [Version.from_string(v) for v in api_versions]
+
+
+def crash_handler():
+    """Implements minimal handling of an exception crashing the application.
+    This function tries to log the exception to a log file and display
+    a minimal crash dialog to the user.
+    This function is supposed to be called from inside an except blog.
+    """
+    # First try to get traceback information and write it to a log file
+    # with minimum chance to fail.
+    import sys
+    from tempfile import NamedTemporaryFile
+    import traceback
+    trace = traceback.format_exc()
+    logfile = None
+    try:
+        with NamedTemporaryFile(suffix='.log', prefix='picard-crash-', delete=False) as f:
+            f.write(trace.encode(errors="replace"))
+            logfile = f.name
+    except:  # noqa: E722,F722 # pylint: disable=bare-except
+        print("Failed writing log file {0}".format(logfile), file=sys.stderr)
+        logfile = None
+
+    # Display the crash information to the user as a dialog. This requires
+    # importing Qt5 and has some potential to fail if things are broken.
+    from PyQt5.QtCore import Qt, QUrl
+    from PyQt5.QtWidgets import QApplication, QMessageBox
+    # assigning QApplication to a variable is required to keep the object alive
+    _unused = QApplication(sys.argv)  # noqa: F841
```

Comment:
Overall it seems to work, but I noticed something: when exiting by closing the message box, the application coredumps.
I think that's because QApplication.quit() is never called.